censustree/iden3tree/tree.go

```diff
@@ -207,23 +217,27 @@ func (t *Tree) Root() []byte {
 	return t.Tree.RootKey().Bytes()
 }
 
-// Dump returns the whole merkle tree serialized in a format that can be used on Import
-func (t *Tree) Dump(root []byte) (keys [][]byte, err error) {
+// Dump returns the whole merkle tree serialized in a format that can be used on Import.
+// TO-DO dump also Values (currently just keys are dumped).
+func (t *Tree) Dump(root []byte) (keys []byte, err error) {
 	rootHash := new(merkletree.Hash)
 	copy(rootHash[:], root)
 	t.updateAccessTime()
 	claims, err := t.Tree.DumpClaims(rootHash)
 	if err != nil {
 		return nil, err
 	}
+	dump := exportData{}
 	for _, c := range claims {
 		cb, err := hex.DecodeString(c)
```

Comment:
would be nice to replace this hex at some point too, e.g. with base64 or just []byte directly. do we have a TODO or issue about it?

Reply:
The dump returned by this function is already []bytes. I suppose you refer to the t.Tree.DumpClaims(). That should be changed upstream on the iden3 merkle tree implementation. We have to update this dependency, actually there exist a branch with this work started but it was abandoned in some point.

Reply:
Ah, gotcha. Maybe we can revive that now, if you think it's worth it.

Reply:
https://gitlab.com/vocdoni/go-dvote/-/tree/feature/snark-int

Reply:
The code is quite old and probably requires quite an effort to update it. We are not using iden3 merkle-tree anywhere in the code right now, it will be used once we integrate zk-SNARKs (few months?). So I'm not sure if its worth it. However if you like to revive it and apply the changes, please do it :+1:

Reply:
Gotcha, I'll give that a go when this is merged.